src/app/domain/components/casebar/casebar.component.ts

```diff
@@ -1,16 +1,26 @@
 import {Component, Input, OnInit} from '@angular/core';
 import 'rxjs/add/operator/filter';
 import {PageDateCaseBar} from '../../models/section_fields';
+import {CaseService} from '../../services/case.service';
 
 @Component({
-        selector: 'app-casebar',
+    selector: 'app-casebar',
     templateUrl: './casebar.component.html',
     styleUrls: ['./casebar.component.scss']
 })
-export class CaseBarComponent {
+export class CaseBarComponent implements OnInit {
 
     @Input() case: PageDateCaseBar;
+    isHidden = false;
 
-    constructor() {}
+    constructor(private caseservice: CaseService) {}
+
+    ngOnInit() {
+        this.caseservice.events$.forEach(event => {
+            if (event === 'hideCasebar') {
```

Comment:
$ on the end means that is an observable can u loop through an observables.

Reply:
yes we can